### PR TITLE
Rename code folder to src, update phpcs rulset to PSR-12 as a base and fix violations

### PR DESCRIPTION
**Pull request by robbieaverill**
There are still two exclusions to the ruleset which require core API changes.

Issue: https://github.com/silverstripe/silverstripe-framework/issues/7991

### CODE REVIEW

File: src/ReportWrapper.php [hunk 1]
```diff
@@ -21,7 +21,7 @@
 
     public function __construct($baseReport)
     {
-        $this->baseReport = is_string($baseReport) ? new $baseReport : $baseReport;
+        $this->baseReport = is_string($baseReport) ? new $baseReport() : $baseReport;
```
Comment (ScopeyNZ) [inline src/ReportWrapper.php]:
```suggestion
        $this->baseReport = is_string($baseReport) ? Injector::inst()->create($baseReport) : $baseReport;
```

Comment (ScopeyNZ) [reply src/ReportWrapper.php]: That might need some spacing updates...

Comment (robbieaverill) [reply src/ReportWrapper.php]: Needs the namespace too- I cant accept from my phone

**Comment by maxime-rainville** [comment]
Is Guy's feedback the only thing holding this from being merge?

**Comment by robbieaverill** [comment]
Updated, sorry this fell off my radar

**Pull request merged**